server/routes/tenants.js

```diff
@@ -11,19 +11,61 @@ const router = express.Router();
 router.get('/tenants/:tenantId', async (req, res, next) => {
   const tenantId = req.params.tenantId;
   try {
-    const foundTenant = await Tenant.findById(tenantId).populate('propertyId');
-    if (foundTenant) {
-      const { address } = foundTenant.propertyId;
-      res.status(StatusCodes.OK).json({ ...foundTenant.toObject(), address });
+    const findtenant = await Tenant.findById(tenantId);
+    if (findtenant.propertyId) {
+      const findTenantWithpropertyId = await Tenant.findById(tenantId).populate('propertyId');
+      if (findTenantWithpropertyId) {
+        const { address } = findTenantWithpropertyId.propertyId;
+        res.status(StatusCodes.OK).json({ ...findTenantWithpropertyId.toObject(), address });
+      } else {
+        res.status(StatusCodes.BAD_REQUEST).json({ error: `Tenant with id ${tenantId} does not exist` });
+      }
     } else {
-      res.status(StatusCodes.BAD_REQUEST).json({ error: `Tenant with id ${tenantId} does not exist` });
+      const customJson = {
```

Comment:
I also don't understand the purpose of returning a custom json. Why return data that does not exist in the DB?